manager/allocator/networkallocator/networkallocator.go

```diff
@@ -151,9 +151,10 @@ func (na *NetworkAllocator) Deallocate(n *api.Network) error {
 
 // ServiceAllocate allocates all the network resources such as virtual
 // IP and ports needed by the service.
-func (na *NetworkAllocator) ServiceAllocate(s *api.Service) (err error) {
+func (na *NetworkAllocator) ServiceAllocate(s *api.Service) error {
+	var err error
```

Comment:
Go back to defining `err` in the function's return value. Otherwise, the `defer` only works properly when code sets this `err`, not some other variable named `err` in an inner scope. It's easy to accidentally break things.